plugins/module_utils/network/sonic/config/bgp_ext_communities/bgp_ext_communities.py

```diff
@@ -369,3 +414,97 @@ def validate_type(self, want):
 
                 new_want.append(cfg)
         return new_want
+
+    def get_replaced_overridden_config(self, want, have, cur_state):
+        commands, requests = [], []
+
+        commands_del, requests_del = [], []
+        commands_add, requests_add = [], []
+
+        for conf in want:
+            name = conf['name']
+            in_have = False
+            for have_conf in have:
+                if have_conf['name'] == name:
+                    in_have = True
+                    if have_conf['type'] != conf['type']:
+                        # If both extended community list are of same name but different types
+                        commands_del.append(have_conf)
+                        commands_add.append(conf)
+                    else:
+                        is_change = is_delete = False
+                        add_conf, delete_conf = {}, {}
+
+                        if have_conf['permit'] != conf['permit']:
+                            is_change = is_delete = True
+
+                        if have_conf['match'] != conf['match']:
+                            is_change = is_delete = True
+
+                        if conf["type"] == "expanded":
+                            members = conf.get('members', {})
+                            if members and conf['members'].get('regex', []):
+                                if have_conf.get('members', {}) and have_conf['members'].get('regex', []):
+                                    if get_diff(have_conf['members']['regex'], members['regex']):
+                                        is_delete = is_change = True
+                            else:
+                                # If there are no members in any extended community list of want, then
+                                # that particular ext community list request to be removed or ignored since
+                                # expanded type needs community-member to exist
+                                if have_conf.get('members', {}) and have_conf['members'].get('regex', []):
+                                    break
```

Comment:
Removed is_delete.
Instead of break, added fail_json message in those failed cases since without community members(For expanded) and attributes(For standard), community-sets can not exist.

Reply:
The revised handling looks better. I have posted an additional comment on the revised code for the "expanded" type handling of this case (although the "standard" type handling looks fine now), so I will mark this issue as "Resolved" to avoid redundancy.